knowledge-base/how-to-add-interactivity-to-a-report-using-action-event-handlers-in-wpf-report-viewer.md

```diff
@@ -0,0 +1,195 @@
+---
+title: How to add interactivity to a report using action event handlers in WPF report viewer
+description: How to add interactivity to a report using action event handlers in WPF report viewer.
+type: how-to
+page_title: How to add interactivity to a report using action event handlers in WPF report viewer
+slug: how-to-add-interactivity-to-a-report-using-action-event-handlers-in-wpf-report-viewer
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting WPF Report Viewer</td>
+		</tr>
+	</tbody>
+</table>
+
+## Desciption
+
+Handle interactive action events in Telerik Report Viewers.  
+  
+## Solution
+
+The report viewers provide handlers for three types of events that are associated with interactive actions – **Executing**, **Enter** and **Leave**. In this article we will show how to use these events to add more interactivity and allow further customization for your reports. We will use the **CSharp.NetFramework.WpfIntegrationDemo** project that is shipped with your Telerik Reporting installation and is part of the **CSharp.ReportExamples.VSxxxx**  solution that can be found at the following path - *C:\Program Files (x86)\Progress\\<Version\>\Examples\CSharp*.   
+  
+The report we chose for this example is part of our **ReportingLibrary** and is called *PopulationDensity*. It displays a *choropleth* based on the world population by countries. We will show how to change the colors and display an informational tooltip for the country under the mouse cursor, using a [Custom](../designing-reports-interactivity-custom-action) action. Additionally, we will show you how to cancel a [NavigateToUrl](../designing-reports-interactivity-hyperlinks) action before it is executed, depending on a confirmation dialog. 
+
+
+- Start Visual Studio, open the solution file and locate the **CSharp.ReportLibrary** project. Navigate to **Choropleth/PopulationDensity.cs** file and open it. Add the following code snippet in the report *constructor* right after the **InitializeComponent()** call:
```

Comment:
As the below solution is rather long and comprehensive format it with bullets and make it a step by step guidance. Also think if we can add some code comments for better visibility of what the code does.